### PR TITLE
[Counterfactual] Add first steps widget

### PR DESCRIPTION
**Pull request by usame-algan**
## What it solves

Part of #3207 

## How this PR fixes it

- Creates a new widget for the dashboard to show first steps
- Creates `StatusProgress` component that can be reused for other widgets

## How to test it

1. Create a new counterfactual safe
2. Observe first steps card visible on the dashboard
3. Complete first steps or deploy safe immediately
4. Observe cards are not visible anymore

## Screenshots
Screenshot 2024-02-07 at 16 55 48

## Checklist
* [ ] I've tested the branch on mobile 📱
* [ ] I've documented how it affects the analytics (if at all) 📊
* [ ] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

**Comment by usame-algan** [comment]
I think we can show this widget to any safe and add a close button in the corner to permanently hide it. cc @liliiaorlenko

File: src/components/dashboard/FirstSteps/index.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+type StatusProgressItems = Array<{
+  name: string
+  completed: boolean
+}>
+
+const StatusProgress = ({ items }: { items: StatusProgressItems }) => {
+  const totalNumberOfItems = items.length
+  const completedItems = items.filter((item) => item.completed)
+  const progress = Math.round((completedItems.length / totalNumberOfItems) * 100)
```
Comment (katspaugh) [inline src/components/dashboard/FirstSteps/index.tsx:29]: These three vars are only needed to compute the `progress` var. Extract them into a `calculateProgress` function?

File: src/components/dashboard/FirstSteps/index.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  const outgoingTransactions = useAppSelector(selectOutgoingTransactions)
+
+  const hasNonZeroBalance = balances && (balances.items.length > 1 || BigInt(balances.items[0]?.balance || 0) > 0)
+  const hasOutgoingTransactions = !!outgoingTransactions && outgoingTransactions.length > 0
+
+  const items = [
+    { name: 'Add funds', completed: hasNonZeroBalance },
+    { name: 'Create your first transaction', completed: hasOutgoingTransactions },
+    { name: 'Safe is ready', completed: hasNonZeroBalance && hasOutgoingTransactions },
+  ]
```
Comment (katspaugh) [inline src/components/dashboard/FirstSteps/index.tsx:61]: I would
* define the names of steps as a sepate enum
* memoize this object

File: src/components/dashboard/FirstSteps/index.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    <>
+      <List disablePadding sx={{ py: 3 }}>
+        {items.map(({ name, completed }) => {
+          return (
+            <StatusItem key={name} completed={completed}>
+              {name}
+            </StatusItem>
+          )
+        })}
+        {completedName && <StatusItem completed={progress === 100}>{completedName}</StatusItem>}
```
Comment (katspaugh) [inline src/components/dashboard/FirstSteps/index.tsx:45]: Aren't all items supposed to be shown at all times? And if not completed it's grey?

**Review by katspaugh — APPROVED** [review]

**Comment by usame-algan** [comment]
@katspaugh I implemented the alternative design with some placeholder values for counterfactual so I would wait to merge this until #3180 is merged.

**Comment by francovenica** [comment]
This ticket was tested and looks good, but with Usame we have decied to merge this ticket along with 3220 and 3214 since these 3 are piceces of a functionality, and it is easier to test all together.

**Pull request merged**